File: libasynDiscordWebHook-PM/src/ialdrich23xx/libasynwebhook/discord/body/embed/components/Thumbnail.php
<?php

declare(strict_types=1);

namespace ialdrich23xx\libasynwebhook\discord\body\embed\components;

use ialdrich23xx\libasynwebhook\discord\body\embed\base\Structure;
use ialdrich23xx\libasynwebhook\discord\body\embed\base\URL;

class Thumbnail extends Structure
{
    use URL;

    public function __construct(
        string $url
    ){
        $this->setUrl($url);
    }

    public static function make(string $url): self
    {
        return new self($url);
    }

    public function build(): bool
    {
        return $this->urlBuild();
    }

    public function toArray(): array
    {
        return $this->urlToArray();
    }

    public function toString(): string
    {
        return "Thumbnail(" . $this->urlToString() .  ")";
    }
}

File: libasynDiscordWebHook-API/src/main/java/com/ialdrich23xx/libasynwebhook/api/discord/body/embed/components/Thumbnail.java
package com.ialdrich23xx.libasynwebhook.api.discord.body.embed.components;

import com.ialdrich23xx.libasynwebhook.api.discord.body.embed.base.Structure;
import com.ialdrich23xx.libasynwebhook.api.discord.body.embed.base.URL;

import java.util.Map;

public class Thumbnail extends Structure implements URL {

    public Thumbnail(String url) {
        this.setUrl(url);
    }

    public static Thumbnail make(String url) {
        return new Thumbnail(url);
    }

    @Override
    public Boolean build() {
        return this.urlBuild();
    }

    @Override
    public Map<String, Object> toArray() {
        return this.urlToArray();
    }

    @Override
    public String toString() {
        return "Thumbnail(" + this.urlToString() +  ")";
    }
}
